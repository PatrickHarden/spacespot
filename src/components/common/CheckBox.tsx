import React from 'react'
import styled from 'styled-components'

const CheckBox = styled(
  (props: {
    children: React.ReactNode
    className?: string
    checked?: boolean
    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void
  }) => {
    return (
      <label className={props.className}>
        {props.children}
        <input
          type="checkbox"
          checked={props.checked}
          onChange={props.onChange}
        />
        <span></span>
      </label>
    )
  },
)`
  display: block;
  position: relative;
  cursor: pointer;
  padding-left: 25px;
  margin: 10px 0;
  input {
    position: absolute;
    opacity 0;
    cursor: pointer;
    width: 0;
    height: 0
  }
  span {
    position: absolute;
    top: 0;
    left: 0;
    height: 15px;
    width: 15px;
    border: 2px solid #B3DBE9;
    margin-top: 1px;
  }
  input:checked ~ span {
    background-color: transparent;
  }

  span:after {
    content: " ";
    position: absolute;
    display: none;
  }

  input:checked ~ span:after {
    display: block;
  }

  span:after {
    left: 5px;
    top: 2px;
    width: 5px;
    height: 8px;
    border: solid black;
    border-width: 0 1px 2px 0;
    transform: rotate(45deg);
  }
`

export default CheckBox
