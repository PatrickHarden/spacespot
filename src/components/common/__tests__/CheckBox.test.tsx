import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import CheckBox from '../CheckBox'

it('renders without crashing ', async () => {
  const wrapper = shallow(<CheckBox>Test</CheckBox>)
  expect(shallowToJson(wrapper)).toMatchSnapshot()
})
