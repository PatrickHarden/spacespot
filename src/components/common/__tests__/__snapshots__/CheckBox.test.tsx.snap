// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing  1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: block;
  position: relative;
  cursor: pointer;
  padding-left: 25px;
  margin: 10px 0;
  input {
    position: absolute;
    opacity 0;
    cursor: pointer;
    width: 0;
    height: 0
  }
  span {
    position: absolute;
    top: 0;
    left: 0;
    height: 15px;
    width: 15px;
    border: 2px solid #B3DBE9;
    margin-top: 1px;
  }
  input:checked ~ span {
    background-color: transparent;
  }

  span:after {
    content: \\" \\";
    position: absolute;
    display: none;
  }

  input:checked ~ span:after {
    display: block;
  }

  span:after {
    left: 5px;
    top: 2px;
    width: 5px;
    height: 8px;
    border: solid black;
    border-width: 0 1px 2px 0;
    transform: rotate(45deg);
  }
",
        ],
      },
      "displayName": "Styled(Component)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Test
</StyledComponent>
`;
